install/openshift/spaship-mongodb.yaml

```diff
@@ -0,0 +1,590 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: spaship-mongodb-template
+metadata:
+  name: spaship-mongodb-template
+  annotations:
+    "description": ""
+    "iconClass": "icon-github"
+    "openshift.io/display-name": "SPAship"
+    "openshift.io/provider-display-name": "SPAship"
+    "openshift.io/support-url": "https://github.com/spaship/spaship"
+    "openshift.io/documentation-url": "https://spaship.io"
+    "tags": "spaship,spa,mongodb"
+objects:
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: ${name}-www
+    spec:
+      accessModes:
+        - ReadWriteMany
+      resources:
+        requests:
+          storage: ${pvc_capacity}
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: ${name}-mongodb
+    spec:
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: ${pvc_capacity}
+  - apiVersion: v1
+    kind: Secret
+    metadata:
+      name: ${name}-mongodb
+    stringData:
+      database-admin-password: "${MONGODB_ADMIN_PASSWORD}"
+      database-name: "${MONGODB_DATABASE}"
+      database-password: "${MONGODB_PASSWORD}"
+      database-user: "${MONGODB_USER}"
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: ${name}-httpd-config
+    data:
+      forbidden.conf: |-
+        <FilesMatch "^spaship\.yaml">
+            Require all denied
+        </FilesMatch>
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: ${name}-httpd-ssi
+    data:
+      index.html: >-
+        <!--#if expr="v('HTTP_COOKIE') =~ /rh_locale=(en|ja|ko|zh_CN)/ && $1 =~
+        /(.*)/" -->
+            <!--#set var="locale" value="$0" -->
+        <!--#else -->
+            <!--#set var="locale" value="en" -->
+        <!--#endif -->
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: ${name}-api-config
+    data:
+      API_PORT: "8080"
+      AUTH__KEYCLOAK__CLIENTID: spaship-reference
+      AUTH__KEYCLOAK__REALM: EmployeeIDP
+      AUTH__KEYCLOAK__URL: https://auth.redhat.com/auth
+      AUTH__KEYCLOAK__ID_PROP: rhatUUID
+      AUTH__KEYCLOAK__PUBKEY: >-
+        MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAtPPUahQ3PJhoCqSsvWTnR/mq4bw4PePpBgjQlRVzUBZwW5scKHbUZxdK9qRhX3fmMOz0VN3H4A+1launCJz56blQFNx9PhZ4vka83qCrNjtlr5lQZsRSuCUO283gJ303fdqQhmrmveVZ34KlUbehcy/m71nP92Aq/inICN5XDmXRDyd/1jDl1oT6XCmEaO67Lv7yPUdfTQR8Dfvw48lnqupFS6dbY3QNO0YHIoanmeX9wao3DLioXBEFY8IsHKim0lAVtlQieWCl5jr9UemrIxzCbIkTwVkjyM4QMocNBhl7zOhfKt8NBBMVAznrdPBX5dUpe96WqCnDUO4UfZY6G6bMux4atI+CJ5NF0fqUlVEE/sgnsjuGmksTnPu+QgM76rzEi61D/lEk9LFvtZr9Iwr1iWm54Ot4rDch8dGi0wvKfTFjsbJDQWenHt/p+9JBpN3DG4b3gAf0YDLNt9l4VCnCQKbLhozy/N74RPZ9t7deSF7r/tl+vBAhrqBluFTauw4qt0KGUhwGtoMGvW3C2EaDoBrVncrEPvipLPl1z4C+FB3CllCyiZczodtCcqYvTdigVOSBnHEb3tYwnyENMNuOmkX07YbFBetrzA+k9vHuVx4owT5TTwAPlcd7ghAmWKQJLJsqivsLlLHDgB9xVtMq20n7UTo8zb3dP8KZ8F0CAwEAAQ==
```

Comment:
Looking at this part, I'm wondering if we should write some kind of script to auto-populate these ConfigMap values into the YAML file somehow.

My reasoning behind this is that these particular set of values are the actual values for our reference implementation. For any layperson wanting to setup SPAship using this template, it might be not outright clear that they would need to update these values here.

Reply:
We had talked this before, resolve it